src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -2685,5 +2652,779 @@ public static int M()
                 Assert.Equal(expected, actual);
             });
         }
+
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/69472")]
+        [InlineData("string", "11")]
+        [InlineData("object", "18")]
+        [InlineData("C", "18")]
```

Comment:
>"18" [](http://example.com/codeflow?start=25&length=4)

Do `C` and `object` have the same type code, but distinct from `string`? Is it a potential issue that the two have the same type code?

Reply:
I'll add a test to make this clearer. `string` has its own `PrimitiveTypeCode`, but `object` and `C` both are `PrimitiveTypeCode.Reference`. The `PrivateImplementationDetails` is an `object[]` for both of those, and it is used to make a `ReadOnlySpan` of the appropriate reference type (either `object` or `C`).